### PR TITLE
chore(rpc): more trace prep

### PR DESCRIPTION
**Pull request by mattsse**
more preparations for tracing impls:

extend EthTransactions trait

trait bounds etc...

### CODE REVIEW

File: crates/rpc/rpc/src/eth/api/transactions.rs [hunk 1]
```diff
@@ -5,32 +5,64 @@ use crate::{
...
+                // settings like basefee/number
+                unimplemented!("support pending state env")
+            }
+            hash_or_num => {
+                let block_hash = self
+                    .client()
+                    .block_hash_for_id(hash_or_num)?
+                    .ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+                let (cfg, env) = self.cache().get_evm_env(block_hash).await?;
+                Ok((cfg, env, block_hash.into()))
```
Comment (rkrasiuk) [inline crates/rpc/rpc/src/eth/api/transactions.rs:61]: is block hash lookup needed? why not just pass `hash_or_num` to the fn?

**Review by rkrasiuk — APPROVED** [review]
lgtm pending comment below

File: crates/rpc/rpc/src/trace.rs [hunk 2]
```diff
@@ -107,13 +113,20 @@ where
     /// Handler for `trace_transaction`
     async fn trace_transaction(
         &self,
-        _hash: H256,
+        hash: H256,
     ) -> Result<Option<Vec<LocalizedTransactionTrace>>> {
+        let (_transaction, at) = match self.eth_api.transaction_by_hash_at(hash).await? {
```
Comment (gakonst) [inline crates/rpc/rpc/src/trace.rs:118]: Getting closer....

File: crates/rpc/rpc/src/eth/api/transactions.rs (hunk 1 above)
Comment (gakonst) [reply crates/rpc/rpc/src/eth/api/transactions.rs:61]: Because it might be a non existent block perhaps?

**Review by gakonst — APPROVED** [review]

**Pull request merged**